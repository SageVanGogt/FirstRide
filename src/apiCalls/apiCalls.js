import { geoKey } from './../apiKey';

const signinUser = async (user) => {
  const url = 'http://localhost:3000/api/users/';
  const init = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, init);
  const profile = await response.json();
  return profile;
}; 

const signupUser = async (user) => {
  const url = 'http://localhost:3000/api/users/new';
  const body = {
    method: "POST",
    body: JSON.stringify(user),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const userRecieved = await response.json();
  return userRecieved;
};

const fetchDestination = async (name) => {
  const url = `http://localhost:3000/api/locations/${name}`;
  const response = await fetch(url);
  const destination = await response.json();
  return destination;
};

const fetchRides = async (destination) => {
  const url = `http://localhost:3000/api/rides/get/${destination}`;
  const body = {
    method: 'POST', 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = await response.json();
  return rides;
};

const fetchPickups = async (location) => {
  const url = `http://localhost:3000/api/pickup/get/${location}`;
  const body = {
    method: "POST", 
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const pickups = await response.json();
  return pickups;
};

const fetchGeocode = async (address) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?address=${address}&key=${geoKey}`;  
  try {
    const response = await fetch(url);
    const locationData = await response.json();
    if (!locationData) {
      return new Error('error')
    }
    return locationData;
  } catch (error) {
    throw error;
  }
}; 

const submitNewRide = async (ride) => {
  const url = `http://localhost:3000/api/rides/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(ride),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const rideId = await response.json();
  return rideId;
};

const submitNewPickup = async (location) => {
  const url = `http://localhost:3000/api/pickup/new`;
  const body = {
    method: 'POST',
    body: JSON.stringify(location),
    headers: {
      "Content-Type": "application/json"        
    }
  };
  const response = await fetch(url, body);
  const pickupId = await response.json();
  return pickupId;
};

const fetchRidesPassengers = async (locationId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/passengers/${locationId}`;
  const body = {
    method: 'POST', 
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const passengers = await response.json();
  return passengers;
};

const postRidesPassengers = async (newPassenger) => {
  const url = `http://localhost:3000/api/rides_passengers/new`;   
  const body = {
    method: "POST",
    body: JSON.stringify(newPassenger),
    headers: {
      'Content-Type': 'application/json'
    }
  };
  const response = await fetch(url, body);
  const ridePassenger = await response.json();
  return ridePassenger;
};

const postNewProfile = async (profile) => {
  const url = `http://localhost:3000/api/profiles/new`;
  const body = {
    method: "POST",
    body: JSON.stringify(profile),
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const profileInfo = await response.json();
  return profileInfo;
};

const removePassengerRide = async (rideId, passengerId, destinationId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/passengers/${passengerId}/destination/${destinationId}`; 
  const body = {
    method: "DELETE",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const deletedId = await response.json();
  return deletedId;
};

const fetchUserRides = async (userId) => {
  const url = `http://localhost:3000/api/rides_passengers/get/rides/${userId}`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const rides = response.json();
  return rides;
};

const fetchRidesFromUser = async (rideId) => {
  const url = `http://localhost:3000/api/rides/${rideId}/get/user/`;
  const body = {
    method: "POST",
    headers: {
      "Content-Type": "application/json"
    }
  };
  const response = await fetch(url, body);
  const ridesInfo = await response.json();
  return ridesInfo;
};

const reverseGeoCode = async (lat, lng) => {
  const url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lng}&key=${geoKey}`;
  const response = await fetch(url);
  const geoData = await response.json();
  return geoData;
}

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewRide,
  submitNewPickup,
  fetchRidesPassengers,
  postRidesPassengers,
  postNewProfile,
  removePassengerRide,
  fetchUserRides,
  fetchRidesFromUser,
  reverseGeoCode
};