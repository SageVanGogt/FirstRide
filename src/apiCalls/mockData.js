export const mockUser = {
  id:1,
  user_id:"456",
  user_img:"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
  bio:"I like to eat",
  rating:"Solid 7"
}