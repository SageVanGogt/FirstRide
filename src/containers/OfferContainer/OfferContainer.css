.offer-container {
  position: absolute;
  z-index: 2;
}