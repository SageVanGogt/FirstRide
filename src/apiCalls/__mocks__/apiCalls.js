import * as MOCK from './../mockData';

const signinUser = jest.fn().mockImplementation(() => 
  Promise.resolve({
    user: [{
    "id":1,
    "user_id":"456",
    "user_img":"http://us.cdn281.fansshare.com/photos/channingtatum/bfhgoi-ccaagvrc-large-fat-44506770.jpg",
    "bio":"I like to eat",
    "rating":"Solid 7"
    }]
  })
);

const signupUser = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1
}));

const fetchDestination = jest.fn().mockImplementation(() => Promise.resolve({
  "id": 1,
  "location_name": "Red Rocks",
  "location_lat_lng": "12341, 123412"
}));

const fetchRides = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockRides));

const fetchPickups = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockPickups));

const fetchGeocode = jest.fn().mockImplementation(() => 
Promise.resolve(MOCK.mockGeoInfo));

const submitNewRide = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const submitNewPickup = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

const postRidesPassengers = jest.fn().mockImplementation(() => 
Promise.resolve({
  id: 1
}));

export {
  signinUser,
  signupUser,
  fetchDestination,
  fetchRides,
  fetchPickups,
  fetchGeocode,
  submitNewPickup,
  submitNewRide,
  postRidesPassengers
}