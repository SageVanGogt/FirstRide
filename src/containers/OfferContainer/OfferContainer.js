import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import * as API from './../../apiCalls/apiCalls';
import * as cleaner from './../../cleaners/cleaners';
import * as actions from './../../actions/pickups';
import './OfferContainer.css';

export class OfferContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location_id: this.props.destination.id,
      driver_id: this.props.user.id,
      car_capacity: '',
      car_type: '',
      date: '',
      time: '',
      street: '',
      city: '',
      state: ''
    }
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  handleSubmitRide = async (event) => {
    event.preventDefault();
    const { 
      location_id,
      driver_id,
      car_capacity,
      car_type,
      date,
      time
    } = this.state
    const rideInfo = {
      location_id,
      driver_id,
      car_capacity, 
      seats_remaining: car_capacity, 
      car_type, 
      date, 
      time, 
    }
    const response = await API.submitNewRide(rideInfo);
    const rideId = response.id;
    await this.handleSubmitPickup(rideId);
    await this.props.loadRides();
  }

  handleSubmitPickup = async (rideId) => {
    const geoLocation = await this.getGeoInfo();
    const pickupInfo = {
      ride_id: rideId,
      location_id: this.state.location_id,
      lat: geoLocation.lat,
      lng: geoLocation.lng,
      isShowing: false
    };
    const newPickup = await API.submitNewPickup(pickupInfo);
    this.props.setNewPickup(newPickup.pickup);
  }

  getGeoInfo = async () => {
    const formattedAddress = this.formatAddress();
    const response = await API.fetchGeocode(formattedAddress);
    const geoLocation = await cleaner.geocodeCleaner(response);
    return geoLocation;
  }

  formatAddress = () => {
    const street = this.state.street.replace(' ', '+');
    const city = this.state.city.replace(' ', '+');
    const state = this.state.state.replace(' ', '+');
    return (
      `${street},+${city},+${state}`
    );
  }

  render() {
    return (
      <div className="offer-container"> 
        <form 
          action="" 
          onSubmit={this.handleSubmitRide}
          className="offer-form">
           <legend className="offer-instructions">
          this is where you input car info 
          please input all the fields
          </legend>
          <button 
            onClick={this.props.handleShowOffer}
            className="offer-back-btn">
            exit
          </button>          
          <div className="offer-inputs-container">
            <div className="car-input">
              <input 
                type="text" 
                name="car_capacity" 
                onChange={this.handleChange} 
                placeholder="car capacity"
                className="offer-input"/>
              <input 
                type="text" 
                name="car_type" 
                onChange={this.handleChange} 
                placeholder="car type"
                className="offer-input"/>
              <input 
                type="text" 
                name="date" 
                onChange={this.handleChange} 
                placeholder="date"
                className="offer-input"/>
              <input 
                type="text" 
                name="time" 
                onChange={this.handleChange} 
                placeholder="time of departure"
                className="offer-input"/>
            </div>
            <div className="address-input">
              <input 
                type="text" 
                name="street" 
                onChange={this.handleChange} 
                placeholder="street"
                className="offer-input"/>
              <input 
                type="text" 
                name="city" 
                onChange={this.handleChange} 
                placeholder="city"
                className="offer-input"/>
              <input 
                type="text" 
                name="state" 
                onChange={this.handleChange} 
                placeholder="state"
                className="offer-input"/>
            </div>
          </div>
          <input 
            type="submit"
            className="offer-submit-btn"/>
        </form>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  destination: state.destination,
  user: state.user
})

export const mapDispatchToProps = (dispatch) => ({
  setNewPickup: (pickup) => dispatch(actions.addSinglePickup(pickup))
})

export default connect(mapStateToProps, mapDispatchToProps)(OfferContainer)