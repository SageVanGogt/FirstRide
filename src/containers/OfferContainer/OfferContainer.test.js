import React from 'react';
import { shallow } from 'enzyme';
import { OfferContainer, mapStateToProps, mapDispatchToProps } from './OfferContainer';
import * as API from './../../apiCalls/apiCalls';
import * as MOCK from './../../apiCalls/mockData';

jest.mock('./../../apiCalls/apiCalls');

describe('OfferContainer', () => {
  let wrapper;
  let mockDestination;
  let mockUser;
  let mockLoadRides;
  let mockSetNewPickup;

  beforeEach(() => {
    mockSetNewPickup = jest.fn();
    mockLoadRides = jest.fn();
    mockDestination = { id: 1 };
    mockUser = { id: 1 };
    wrapper = shallow(<OfferContainer 
      user={mockUser}
      destination={mockDestination}
      loadRides={mockLoadRides}
      setNewPickup={mockSetNewPickup}
    />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('handleChange', () => {

    it('should update the state when changes are made to inputs', () => {
      let mockEvent = {
        target: {
          name: 'car_type',
          value: 'sedan'
        }
      }
      let expected = 'sedan'
      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state('car_type')).toEqual(expected);
    })
  })

  describe('handleSubmitRide', () => {
    it('should call submitNewRide with correct params', async () => {
      let expected = {
        location_id: 1,
        driver_id: 1,
        car_capacity: '',
        seats_remaining: '',
        car_type: '',
        date: '',
        time: ''
      }
      let mockEvent = {
        preventDefault: jest.fn()
      }
      await wrapper.instance().handleSubmitRide(mockEvent);

      expect(API.submitNewRide).toHaveBeenCalledWith(expected)
    })

    it('should call loadRides on handle submit', async () => {
      let mockEvent = {
        preventDefault: jest.fn()
      };

      await wrapper.instance().handleSubmitRide(mockEvent);

      expect(mockLoadRides).toHaveBeenCalled();
    })
  })

  describe('handleSubmitPickup', () => {
    let mockRideId;

    beforeEach(() => {
      mockRideId = 1;
    })

    it('should call submitNewPickup with the correct params', async () => {
      let expected = {
        ride_id: 1,
        location_id: 1,
        lat: 39.7594866,
        lng: -104.9994026,
        isShowing: false
      };

      await wrapper.instance().handleSubmitPickup(mockRideId);

      expect(API.submitNewPickup).toHaveBeenCalledWith(expected);
    })

    it('should call setNewPickup with the correct params', async () => {
      let expected = {id: 1};
      await wrapper.instance().handleSubmitPickup(mockRideId);
      
      expect(mockSetNewPickup).toHaveBeenCalledWith(expected);
    })
  })

  describe('getGeoInfo', () => {
    
    it('should call fetchGeoCode with the correct params', async () => {
      let expected = ',+,+'

      await wrapper.instance().getGeoInfo();

      expect(API.fetchGeocode).toHaveBeenCalledWith(expected)
    })
  })

  describe('formatAddress', () => {

    it('should return an address with no spaces', () => {
      let expected = '2600+fairview,+seattle,+WA'
      wrapper.setState({
        street: '2600 fairview',
        city: 'seattle', 
        state: 'WA'
      })
      let actual = wrapper.instance().formatAddress();
      
      expect(actual).toEqual(expected);
    })
  });

  describe('mapStateToProps', () => {

    it('should map the destination to props', () => {
      let mockState = {
        destination: {
          id: 1
        },
        user: {
          id: 1
        }
      };
      let expected = { id: 1 };
      let mappedProps = mapStateToProps(mockState);
      let actual = mappedProps.destination;

      expect(actual).toEqual(expected);
    })

    it('should map the destination to props', () => {
      let mockState = {
        destination: {
          id: 1
        },
        user: {
          id: 1
        }
      };
      let expected = { id: 1 };
      let mappedProps = mapStateToProps(mockState);
      let actual = mappedProps.user;

      expect(actual).toEqual(expected);
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch for setNewPickup with the correct params', () => {
      let mockPickup = {};
      let mockDispatch = jest.fn();
      let mappedProps = mapDispatchToProps(mockDispatch);
      let expected = {
        type: "ADD_NEW_PICKUP",
        pickup: mockPickup
      }
      mappedProps.setNewPickup(mockPickup);

      expect(mockDispatch).toHaveBeenCalledWith(expected);
    })
  })
})