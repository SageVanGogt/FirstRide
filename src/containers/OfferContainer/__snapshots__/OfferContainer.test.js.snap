// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OfferContainer
    destination={
        Object {
            "id": 1,
          }
    }
    user={
        Object {
            "id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        action=""
        onSubmit={[Function]}
>
        <input
                name="car_capacity"
                onChange={[Function]}
                placeholder="car capacity"
                type="text"
        />
        <input
                name="car_type"
                onChange={[Function]}
                placeholder="car type"
                type="text"
        />
        <input
                name="date"
                onChange={[Function]}
                placeholder="date"
                type="text"
        />
        <input
                name="time"
                onChange={[Function]}
                placeholder="time of departure"
                type="text"
        />
        <input
                name="street"
                onChange={[Function]}
                placeholder="street"
                type="text"
        />
        <input
                name="city"
                onChange={[Function]}
                placeholder="city"
                type="text"
        />
        <input
                name="state"
                onChange={[Function]}
                placeholder="state"
                type="text"
        />
        <input
                type="submit"
        />
        <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/rides"
        >
                Back
        </NavLink>
</form>,
      "className": "offer-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "",
        "children": Array [
          <input
            name="car_capacity"
            onChange={[Function]}
            placeholder="car capacity"
            type="text"
/>,
          <input
            name="car_type"
            onChange={[Function]}
            placeholder="car type"
            type="text"
/>,
          <input
            name="date"
            onChange={[Function]}
            placeholder="date"
            type="text"
/>,
          <input
            name="time"
            onChange={[Function]}
            placeholder="time of departure"
            type="text"
/>,
          <input
            name="street"
            onChange={[Function]}
            placeholder="street"
            type="text"
/>,
          <input
            name="city"
            onChange={[Function]}
            placeholder="city"
            type="text"
/>,
          <input
            name="state"
            onChange={[Function]}
            placeholder="state"
            type="text"
/>,
          <input
            type="submit"
/>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/rides"
>
            Back
</NavLink>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "car_capacity",
            "onChange": [Function],
            "placeholder": "car capacity",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "car_type",
            "onChange": [Function],
            "placeholder": "car type",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "date",
            "onChange": [Function],
            "placeholder": "date",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "time",
            "onChange": [Function],
            "placeholder": "time of departure",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "street",
            "onChange": [Function],
            "placeholder": "street",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "city",
            "onChange": [Function],
            "placeholder": "city",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "state",
            "onChange": [Function],
            "placeholder": "state",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Back",
            "to": "/rides",
          },
          "ref": null,
          "rendered": "Back",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          action=""
          onSubmit={[Function]}
>
          <input
                    name="car_capacity"
                    onChange={[Function]}
                    placeholder="car capacity"
                    type="text"
          />
          <input
                    name="car_type"
                    onChange={[Function]}
                    placeholder="car type"
                    type="text"
          />
          <input
                    name="date"
                    onChange={[Function]}
                    placeholder="date"
                    type="text"
          />
          <input
                    name="time"
                    onChange={[Function]}
                    placeholder="time of departure"
                    type="text"
          />
          <input
                    name="street"
                    onChange={[Function]}
                    placeholder="street"
                    type="text"
          />
          <input
                    name="city"
                    onChange={[Function]}
                    placeholder="city"
                    type="text"
          />
          <input
                    name="state"
                    onChange={[Function]}
                    placeholder="state"
                    type="text"
          />
          <input
                    type="submit"
          />
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    to="/rides"
          >
                    Back
          </NavLink>
</form>,
        "className": "offer-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "",
          "children": Array [
            <input
              name="car_capacity"
              onChange={[Function]}
              placeholder="car capacity"
              type="text"
/>,
            <input
              name="car_type"
              onChange={[Function]}
              placeholder="car type"
              type="text"
/>,
            <input
              name="date"
              onChange={[Function]}
              placeholder="date"
              type="text"
/>,
            <input
              name="time"
              onChange={[Function]}
              placeholder="time of departure"
              type="text"
/>,
            <input
              name="street"
              onChange={[Function]}
              placeholder="street"
              type="text"
/>,
            <input
              name="city"
              onChange={[Function]}
              placeholder="city"
              type="text"
/>,
            <input
              name="state"
              onChange={[Function]}
              placeholder="state"
              type="text"
/>,
            <input
              type="submit"
/>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              to="/rides"
>
              Back
</NavLink>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "car_capacity",
              "onChange": [Function],
              "placeholder": "car capacity",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "car_type",
              "onChange": [Function],
              "placeholder": "car type",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "date",
              "onChange": [Function],
              "placeholder": "date",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "time",
              "onChange": [Function],
              "placeholder": "time of departure",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "street",
              "onChange": [Function],
              "placeholder": "street",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "city",
              "onChange": [Function],
              "placeholder": "city",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "state",
              "onChange": [Function],
              "placeholder": "state",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Back",
              "to": "/rides",
            },
            "ref": null,
            "rendered": "Back",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
